runtime-transaction/benches/process_compute_budget_instructions.rs

```diff
@@ -0,0 +1,142 @@
+                ))
+                .is_ok())
+            })
+        });
+    });
+}
+
+fn bench_process_compute_budget_instructions_no_builtins(c: &mut Criterion) {
+    c.bench_function("No builtins", |bencher| {
+        let ixs: Vec<_> = (0..4)
+            .map(|_| {
+                Instruction::new_with_bincode(DUMMY_PROGRAM_ID.parse().unwrap(), &0_u8, vec![])
+            })
+            .collect();
+        let tx = build_sanitized_transaction(&Keypair::new(), &ixs);
+        bencher.iter(|| {
+            (0..NUM_TRANSACTIONS_PER_ITER).for_each(|_| {
+                assert!(process_compute_budget_instructions(black_box(
+                    tx.message().program_instructions_iter()
+                ))
+                .is_ok())
+            })
+        });
+    });
+}
+
+fn bench_process_compute_budget_instructions_compute_budgets(c: &mut Criterion) {
+    c.bench_function("Only compute-budget instructions", |bencher| {
+        let ixs = vec![
+            ComputeBudgetInstruction::request_heap_frame(40 * 1024),
+            ComputeBudgetInstruction::set_compute_unit_limit(u32::MAX),
+            ComputeBudgetInstruction::set_compute_unit_price(u64::MAX),
+            ComputeBudgetInstruction::set_loaded_accounts_data_size_limit(u32::MAX),
+        ];
+        let tx = build_sanitized_transaction(&Keypair::new(), &ixs);
+        bencher.iter(|| {
+            (0..NUM_TRANSACTIONS_PER_ITER).for_each(|_| {
+                assert!(process_compute_budget_instructions(black_box(
+                    tx.message().program_instructions_iter()
+                ))
+                .is_ok())
+            })
+        });
+    });
+}
+
+fn bench_process_compute_budget_instructions_builtins(c: &mut Criterion) {
+    c.bench_function("Only builtins", |bencher| {
+        let ixs = vec![
+            Instruction::new_with_bincode(solana_sdk::bpf_loader::id(), &0_u8, vec![]),
+            Instruction::new_with_bincode(solana_sdk::secp256k1_program::id(), &0_u8, vec![]),
+            Instruction::new_with_bincode(
+                solana_sdk::address_lookup_table::program::id(),
+                &0_u8,
+                vec![],
+            ),
+            Instruction::new_with_bincode(solana_sdk::loader_v4::id(), &0_u8, vec![]),
+        ];
+        let tx = build_sanitized_transaction(&Keypair::new(), &ixs);
+        bencher.iter(|| {
+            (0..NUM_TRANSACTIONS_PER_ITER).for_each(|_| {
+                assert!(process_compute_budget_instructions(black_box(
+                    tx.message().program_instructions_iter()
+                ))
+                .is_ok())
+            })
+        });
+    });
+}
+
+fn bench_process_compute_budget_instructions_mixed(c: &mut Criterion) {
+    c.bench_function("Mixed instructions", |bencher| {
+        let payer_keypair = Keypair::new();
+        let mut ixs: Vec<_> = (0..128)
+            .map(|_| {
+                Instruction::new_with_bincode(DUMMY_PROGRAM_ID.parse().unwrap(), &0_u8, vec![])
```

Comment:
why give all the ixs a single byte for data?

Reply:
those benchmarks only look at program_id for non-builtin (including compute-budget), having data or not doesn't impact results. Having `0` as data is result of copy/paste

Reply:
a4448527f230213a9a0d3601cf089939af077ba0